src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/DefaultFormWithBody.razor

```diff
@@ -3,23 +3,18 @@
 
 <h2>Default form with body</h2>
 
-<EditForm
-    EditContext="_editContext"
-    method="POST"
-    OnValidSubmit="@(() => { _ = FormData.Entries.TryGetValue("firstName", out var greeting) ? _greeting = (string)greeting : ""; })">
+<EditForm EditContext="_editContext" method="POST">
     <input id="firstName" name="firstName" />
     <input id="send" type="submit" value="Send" />
 </EditForm>
 
-@if (_greeting != null)
+@if (FirstName != null)
 {
-    <p id="pass"> Hello @(_greeting)!</p>
+    <p id="pass">Hello @FirstName!</p>
 }
 
 @code{
-    string? _greeting = null;
-
-    [Inject] FormDataProvider FormData { get; set; }
+    [SupplyParameterFromForm] string? FirstName { get; set; }
```

Comment:
Is there now no way to get to the raw form data?

Reply:
We can probably delete this test, since all this was validating was accessing the raw data without binding involved.

Reply:
There's already a separate test that covers binding.

Reply:
> We can probably delete this test, since all this was validating was accessing the raw data without binding involved.

Done

> Is there now no way to get to the raw form data?

I was hoping we could avoid having a separate API for that, and have a single explanation that all raw incoming HTTP request data (form post values, raw headers, cookies, etc) is available through `IHttpContextAccessor` in the short term, and in the longer term we may expose a more idiomatic solution that covers them all (e.g., `[FromHttpContext] public HttpRequest Request { get; set; }`).

If you think we definitely need this we can consider reintroducing it but I'd be inclined to not do so for now if we can avoid it.

Reply:
No, that's fine with me.

I am just thinking mainly of class libraries, as I believe that will be a blocker, won't it?

Reply:
I'm not sure. What scenario would be blocked?

Reply:
I assume we would do something like `HttpRequest.Form`, instead, isn't it? That requires a framework reference to Microsoft.AspNetCore.App, which is not available in webassembly. And since we can't multi-target, that prevents libraries from targeting both Server/Webassembly within the same package.

Reply:
I see what you mean. It sounds like the question can be applied more broadly to what it means to reference raw HTTP concepts from packages that can also target WebAssembly, rather than being only a point about form data. It does seem like people would have to either multitarget or have an abstraction with a framework-referencing-provider installed in the server project.

If in the short term you're OK with not special-casing a solution for form data, then perhaps longer term we can solve this for all `HttpContext` data.